### DISCUSSION TITLE
GitHub Actions (Beta) support in GitHub Enterprise?

### DISCUSSION THREAD

**Discussion by tedmiston** [community/community#25434] [Enterprise] [answered]
<p>Hi - I’m wondering if there’s an ETA for when GitHub Actions will land in GitHub Enterprise?</p>
<p>I’m interested in using Actions in an existing GitHub Enterprise install (self-hosted), even in its current beta form if possible.</p>
<p>Best,</p>
<p>Taylor</p>

**Comment by tedmiston** [comment]
<p>The marketing site mentions GitHub Actions on the features page, though it is using the old HCL syntax from the first version.</p>
<p></p><div class="lightbox-wrapper"><a class="lightbox" href="https://aws1.discourse-cdn.com/github/original/2X/1/154ea332746c413bbef2c5223e97c63ddd8da170.png" data-download-href="/uploads/short-url/32uvqBEMhUwuqyVEmlu1BdDiT2o.png?dl=1" title="Screen Shot 2019-09-18 at 5.50.14 PM.png" rel="nofollow noopener"><img src="https://user-images.githubusercontent.com/366688/181091103-3eab3329-3291-4000-ab77-644843a57f04.png" alt="Screen Shot 2019-09-18 at 5.50.14 PM.png" data-base62-sha1="32uvqBEMhUwuqyVEmlu1BdDiT2o" width="690" height="441" srcset="https://aws1.discourse-cdn.com/github/optimized/2X/1/154ea332746c413bbef2c5223e97c63ddd8da170_2_690x441.png, https://aws1.discourse-cdn.com/github/optimized/2X/1/154ea332746c413bbef2c5223e97c63ddd8da170_2_1035x661.png 1.5x, https://aws1.discourse-cdn.com/github/optimized/2X/1/154ea332746c413bbef2c5223e97c63ddd8da170_2_1380x882.png 2x" data-small-upload="https://aws1.discourse-cdn.com/github/optimized/2X/1/154ea332746c413bbef2c5223e97c63ddd8da170_2_10x10.png"><div class="meta">
<svg class="fa d-icon d-icon-far-image svg-icon" aria-hidden="true"><use xlink:href="#far-image"></use></svg><span class="filename">Screen Shot 2019-09-18 at 5.50.14 PM.png</span><span class="informations">1608×1029 260 KB</span><svg class="fa d-icon d-icon-discourse-expand svg-icon" aria-hidden="true"><use xlink:href="#discourse-expand"></use></svg>
</div></a></div>However, I can’t find Actions mentioned anywhere in the release notes.
<p><a href="https://enterprise.github.com/releases/" rel="nofollow noopener">https://enterprise.github.com/releases/</a></p>

**Comment by AndreaGriffiths11** [comment]
<p>Hi <a class="mention" href="/tedmiston">@tedmiston</a>, Thank you for being here, I can’t give you a specific date that GitHub Actions will be available in a release version of GitHub Enterprise, but we have heard this request from other customers and we hope that it will be included soon.</p>

**Comment by tedmiston** [comment]
<p><a class="mention" href="/andreagriffiths11">@andreagriffiths11</a> Thank you for the quick reply.  Can you clarify from the screenshot above if the old version of Actions etc is available in Enterprise today?  From what I can tell the feature is not in there when I look for it, so I was confused to see screenshots of it and it listed on the Enterprise website.</p>

**Comment by chrispat** [comment]
<p>Actions currently works in GitHub Enterprise Cloud but is not avaialble in GitHub Enterprise Server.  It is on the roadmap to offer it as part of the Server but that will not happen until sometime next calendar year.</p>

**Comment by vincentclaes** [comment]
<p>Hi Chris, is there a possibility to stay informed on any progress of Actions being made available for the self hosted github enterprise?</p>

**Comment by paulhoke** [comment]
<p>Hi <a class="mention" href="/chrispat">@chrispat</a> ,</p>
<p>We are <strong>very much</strong> interested un GitHub Actions in the Enterprise Server release as well. I was a bit surprised that paying customers aren’t getting this already. Can you please give any updates wwhen this Actions will be available in Enterprise?</p>
<p>Thank you!</p>
<p>Paul</p>

**Comment by painhardcore** [comment]
<p>Hi,<br>
What the current status of this? When in 2020 we can expect this?<br>
Will be great to have ETA :slight_smile: or any news </p>

**Comment by metinsenturk** [comment]
<p>Hi!</p>
<p>I was wondering about the status of this feature as well. GitHub actions is a nice tool to have in self-hosted servers as well. </p>
<p>If you could provide a timeline on this, it would be nice to see.</p>
<p>Regards,</p>

**Comment by converge** [comment]
<p>Looking forward to having it as well, it works pretty well and will be a big addition to my team.</p>

**Comment by AlanOuyang** [comment]
<p>Any update on this one. Any hope of having this feature on the server this year? </p>

**Comment by ghost** [comment]
<p>What’s the update on this?</p>

**Comment by tino-teuber** [comment]
<p>Hi Chris, meanwhile, is there any updated information regarding a release of “Actions” on Github Enterprise Server?</p>

**Comment by ghost** [comment]
<p>Hi,</p>
<p>How this is marked as SOLVED if there is still no Github Actions for Github Server Enterprise? What’s the timeline on it? How can we stay up-to-date about it? Please bear in mind that Github Server Enterprise customers are paying for the service and the minimum expectation is to have the same level of functionalities of users using the free version of the service.</p>
<p>Gitlab is nailing this since a while and although enterprises move slower and tend to avoid migrations, this one (migrating from GitHub Enterprise Server to Gitlab Enterprise Server) will be inevitable if things keep the way it is.</p>
<p>Really looking forward to having visibility about the timelines and plan of action to have Github Actions for server enterprise.</p>
<p>Thanks.</p>

**Comment by Kicker758** [comment]
<p>Good day all,</p>
<p>Has any progress been made on adding actions to our self hosted enterprise servers? My team is ready to start using this feature.</p>

**Comment by HerrmannHinz** [comment]
<p>hello community, hello github,</p>
<p>+1</p>
<p>any ETA? or shall we keep using jenkins/travis/whatever?</p>
<p>thanks in ahead</p>

**Comment by malixsys** [comment]
<p>Cf. This post</p>
<aside class="onebox whitelistedgeneric">
  <header class="source">
      <img src="https://github.blog/wp-content/uploads/2019/01/cropped-github-favicon-512.png?fit=32%2C32" class="site-icon" width="32" height="32">
      <a href="https://github.blog/2020-04-22-github-actions-community-momentum-enterprise-capabilities-and-developer-improvements/" target="_blank" rel="nofollow noopener" title="05:00PM - 22 April 2020">The GitHub Blog – 22 Apr 20</a>
  </header>
  <article class="onebox-body">
    <div class="aspect-image" style="--aspect-ratio:690/362;"><img src="https://github.blog/wp-content/uploads/2019/08/DL-V2-LinkedIn_FB.png?fit=1200%2C630" class="thumbnail" width="690" height="362"></div>

<h3><a href="https://github.blog/2020-04-22-github-actions-community-momentum-enterprise-capabilities-and-developer-improvements/" target="_blank" rel="nofollow noopener">GitHub Actions: Community momentum, enterprise capabilities, and developer...</a></h3>

<p>GitHub Actions continues its community momentum and ships new features for enterprises and developers.</p>


  </article>
  <div class="onebox-metadata">
    
    
  </div>
  <div style="clear: both"></div>
</aside>

**Comment by KaiSforza** [comment, marked as answer]
To anyone watching this, it appears to have [become a beta feature](https://resources.github.com/beta-signup/) in the enterprise server offerings [here](https://docs.github.com/en/enterprise-server@3.7/admin/enterprise-management/updating-the-virtual-machine-and-physical-resources/upgrading-github-enterprise-server)

_Comment Edited by Staff_

**Comment by MichaelZhouW** [comment]
<p>Did you manage to get the beta version ?</p>

**Comment by KaiSforza** [comment]
<p><a class="mention" href="/zwcore">@ZWcore</a></p>
<p>Yes, it’s just a matter of getting signed up for it, you’ll probably have to talk to an account manager or something to actually get access. If you have a low power box running your GHES instance, it requires some…probably not supported changes to get it to work on a limited basis, or just upgrade your system with 4-8 cores and 32+gb of memory.</p>
